src/codeu/chat/common/RsaAesSha.java

```diff
@@ -0,0 +1,140 @@
+package codeu.chat.common;
+
+import codeu.chat.util.Logger;
+import codeu.chat.common.Encryption;
+import java.security.*;
+import javax.crypto.*;
+import javax.crypto.spec.SecretKeySpec;
+
+
+public class RsaAesSha implements Encryption{
+
+  private final static Logger.Log LOG = Logger.newLog(RsaAesSha.class);
+
+  private Key publicKey;
+  private Key privateKey;
+  private Cipher rsa;
+  private Cipher aes;
+  private SecretKeySpec symmetricKey;
+  private MessageDigest md;
+
+  public RsaAesSha() {
+    try {
+      KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA");
+      kpg.initialize(2048);
+      KeyPair kp = kpg.genKeyPair();
+      publicKey = kp.getPublic();
+      privateKey = kp.getPrivate();
+    } catch(Exception ex) {
```

Comment:
A couple of things on this:

1. In general, it is clearer to the reader if you explicitly state the types of exception, rather than just a catch-all "Exception"
2. Will log.error exit out? If not you might consider throwing the exception to the client